static/js/publisher/form/multiselect.js

```diff
@@ -0,0 +1,376 @@
+  }
+
+  sortByName(a, b) {
+    return a.name.localeCompare(b.name);
+  }
+
+  /**
+   * Remove an item from the 'selected' list.
+   * Add it back into the 'values' list.
+   */
+  removeItem(e) {
+    const key = e.target.parentNode.dataset.key;
+    const toRemove = this.state.selected.filter(item => item.key === key)[0];
+    const newValues = this.state.values.slice(0);
+    newValues.push(toRemove);
+    newValues.sort(this.sortByName);
+    const newSelected = this.state.selected.filter(item => item.key !== key);
+
+    this.setState({
+      values: newValues,
+      selected: newSelected,
+      // Preserve the filtered list in the dropdown
+      searchResults: this.filterByTerm(newValues)
+    }, this.updateInputValue);
+  }
+
+  /**
+   * If an element is clicked, remove the highlightedOption before adding
+   * the item to the 'selected' list.
+   *
+   * @param event
+   */
+  clickItem(event) {
+    this.setState({
+      highlightedOption: null
+    });
+    this.addItem(event);
+  }
+
+  /**
+   * Add an item to the 'selected' list.
+   * Remove it from the 'values' list.
+   */
+  addItem(e) {
+    // Get the key from the 'data-key' attribute of the element
+    const key = e.target.dataset.key;
+    // Get the object based on the key
+    const toAdd = this.state.values.filter(item => item.key === key)[0];
+    const newSelected = this.state.selected.slice(0);
+    newSelected.push(toAdd);
+    const newValues = this.state.values.filter(item => item.key !== key);
+    this.searchInput.value = '';
+
+    this.setState({
+      values: newValues,
+      selected: newSelected,
+      searchTerm: '',
+      // Preserve the filtered list in the dropdown
+      searchResults: newValues
+    }, this.updateInputValue);
+  }
+
+  /**
+   * Filter a list by term
+   *
+   * @param values
+   * @param searchTerm
+   * @returns {{name: string, key: string}[]}
+   */
+  filterByTerm(values, searchTerm) {
+    searchTerm = searchTerm || this.state.searchTerm;
+    return values.filter(item => item.name.toLowerCase().startsWith(searchTerm.toLowerCase()));
+  }
+
+  /**
+   * Filter the 'values' list by the search term.
+   */
+  search(e) {
+    const searchTerm = e.target.value;
+    const searchResults = this.filterByTerm(this.state.values, searchTerm);
+    let highlighted = this.state.highlightedOption;
+    if (highlighted) {
+      if (highlighted > searchResults.length - 1) {
+        highlighted = searchResults.length - 1;
+      }
+      if (highlighted < 0) {
+        highlighted = 0;
+      }
+    }
+
+    this.setState({
+      searchResults: searchResults,
+      searchTerm: searchTerm,
+      showSearch: true,
+      highlightedOption: highlighted
+    });
+  }
+
+  /**
+   * Update the original input value and dispatch a change event to the input and form.
+   */
+  updateInputValue() {
+    this.props.input.setAttribute('value', this.state.selected.map(item => item.key).join(', '));
```

Comment:
Wouldn't `this.props.input.value  = ...` do the same thing? And additionaly it should also trigger change event. There really shouldn't be a need to trigger change event manually on a form.

Reply:
aah the magical `new Event('change', { 'bubbles': true });` :+1: thanks